File: share-code/bai-tap-lap-trinh/python/array/61.py
arr     = [1, 2, 3, 2]

# Cách 1:
arr_odd  = []
arr_even = []

for item in arr:
    if item % 2 == 0:
        arr_even.append(item)
    else:
        arr_odd.append(item)

arr_odd  = list(set(arr_odd))
arr_even = list(set(arr_even))

# Cách 2:
# arrOdd = set(item for item in arr if item % 2 != 0)
# arrEven = set(item for item in arr if item % 2 == 0)

# Cách 3:
# arrOdd  = set()
# arrEven = set()

# for item in arr:
#     if item % 2 == 0:
#         arrEven.add(item)
#     else:
#         arrOdd.add(item)

print("arrOdd =", arr_odd)
print("arrEven =", arr_even)

File: share-code/bai-tap-lap-trinh/cpp/06-mang/61.cpp
#include <iostream>
#include <set>
using namespace std;

int main()
{
    int arrInt[] = {1, 2, 3, 2, 1, 4, 5};
    int n = sizeof(arrInt) / sizeof(arrInt[0]);

    set<int> evenSet;
    set<int> oddSet;

    for (int number : arrInt)
    {
        if (number % 2 == 0)
        {
            evenSet.insert(number);
        }
        else
        {
            oddSet.insert(number);
        }
    }

    int arrEven[evenSet.size()];
    int arrOdd[oddSet.size()];

    int idx = 0;
    string resultEven = "";
    for (auto it = evenSet.begin(); it != evenSet.end(); it++)
    {
        string number = to_string(*it);
        string sign = idx == 0 ? "" : ",";
        resultEven += sign + number;
        arrEven[idx++] = *it;
    }

    idx = 0;
    string resultOdd = "";
    for (auto it = oddSet.begin(); it != oddSet.end(); it++)
    {
        string number = to_string(*it);
        string sign = idx == 0 ? "" : ",";
        resultOdd += sign + number;
        arrOdd[idx++] = *it;
    }

    cout << "arrOdd = [" << resultOdd << "]" << endl;
    cout << "arrEven = [" << resultEven << "]" << endl;

    return 0;
}